File: aula5/app/src/main/java/com/example/aula4/MainActivity.java
package com.example.aula4;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import com.google.firebase.Firebase;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;

public class MainActivity extends AppCompatActivity {

    private EditText editTextText, editTextText2;
    private Button button;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);

        //Referencia os PlainTexts e o Button
        editTextText = findViewById(R.id.editTextText);
        editTextText2 = findViewById(R.id.editTextText2);
        button = findViewById(R.id.button);



        // Firebase Database
        FirebaseDatabase database = FirebaseDatabase.getInstance();
        DatabaseReference myRef = database.getReference("users");

        //Listener para o botão
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onCLick(View v) {
                //Capturar os valores dos PlainTexts
                String name = editTextText.getText().toString();
                String email = editTextText2.getText().toString();

                //Criar um novo objeto User com os valores capturados
                User user = new User(name, email);


                //Armazenar os dados no Firebase
                myRef.push().setValue(user);


            }
        });


        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });
    }
}

File: RecyclerView/RecyclerView/app/src/main/java/com/test/recyclerview/MainActivity.kt
package com.test.recyclerview

import android.os.Bundle
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)

        // Dados a serem exibidos no RecyclerView
        val countries = listOf("Brasil", "Estados Unidos", "França", "Alemanha", "Japão")

        // Informações básicas sobre cada país. Mapeia o nome do país com suas informações.
        val countryInfo = mapOf(
            "Brasil" to "Capital: Brasília\nPopulação: 213 milhões\nContinente: América do Sul",
            "Estados Unidos" to "Capital: Washington, D.C.\nPopulação: 331 milhões\nContinente: América do Norte",
            "França" to "Capital: Paris\nPopulação: 67 milhões\nContinente: Europa",
            "Alemanha" to "Capital: Berlim\nPopulação: 83 milhões\nContinente: Europa",
            "Japão" to "Capital: Tóquio\nPopulação: 125 milhões\nContinente: Ásia"
        )

        // Inicializar o RecyclerView
        val recyclerView: RecyclerView = findViewById(R.id.recyclerView)

        // Configurar o LayoutManager (lista vertical)
        recyclerView.layoutManager = LinearLayoutManager(this)

        // Configurar o Adapter, passando a lista de países e as informações de cada país
        recyclerView.adapter = MyAdapter(countries, countryInfo)

        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }
    }
}